lib/client.js

```diff
@@ -728,6 +730,15 @@ Request.prototype._end = function() {
     }
   }
 
+  if(xhr.upload){
+    // upload timeout it's wokrs only if deadline timeout is off
+      if (this._uploadTimeout && !this._uploadTimeoutTimer) {
+          this._uploadTimeoutTimer = setTimeout(() => {
```

Comment:
I suggest assigning that callback to a property and using the property when resetting the timer to avoid recreating the closure frequently.

Reply:
ok 👍